nav2_core/include/nav2_core/local_planner_base.hpp

```diff
@@ -0,0 +1,119 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef NAV2_CORE__LOCAL_PLANNER_BASE_HPP_
+#define NAV2_CORE__LOCAL_PLANNER_BASE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "tf2_ros/transform_listener.h"
+#include "nav2_util/lifecycle_node.hpp"
+#include "pluginlib/class_loader.hpp"
+
+#include "nav_2d_msgs/msg/pose2_d_stamped.hpp"
+#include "nav_2d_msgs/msg/twist2_d_stamped.hpp"
+#include "nav_2d_msgs/msg/path2_d.hpp"
+
+namespace nav2_core
+{
+
+/**
+ * @class LocalPlanner
+ * @brief planner interface that will be inherited by all local planners
+ */
+class LocalPlanner : public nav2_util::LifecycleHelperInterface
+{
+public:
+  using Ptr = std::shared_ptr<nav2_core::LocalPlanner>;
+
+  /**
+   * @brief Constructor that brings up pluginlib loaders
+   */
+
+  virtual void initialize(nav2_util::LifecycleNode::SharedPtr,
+    std::shared_ptr<nav2_costmap_2d::Costmap2DROS>) = 0;
+  /**
+   * @brief Virtual destructor
+   */
+  virtual ~LocalPlanner() {}
+
+  /**
+   * @brief Sets the global goal for this local planner.
+   * @param goal_pose The Goal Pose
+   */
+
+  // virtual void setGoalPose(const nav_2d_msgs::msg::Pose2DStamped& goal_pose) = 0;
+
+  /**
+   * @brief local setPlan - Sets the global plan
+   * @param path The global plan
+   */
+  virtual void setPlan(const nav_2d_msgs::msg::Path2D & path) = 0;
+
+  /**
+   * @brief local_planner_base computeVelocityCommands - calculates the best command given the current pose and velocity
+   *
+   * It is presumed that the global plan is already set.
+   *
+   * This is mostly a wrapper for the protected computeVelocityCommands
+   * function which has additional debugging info.
+   *
+   * @param pose Current robot pose
+   * @param velocity Current robot velocity
+   * @return The best command for the robot to drive
+   */
+  virtual nav_2d_msgs::msg::Twist2DStamped computeVelocityCommands(
+    const nav_2d_msgs::msg::Pose2DStamped & pose,
+    const nav_2d_msgs::msg::Twist2D & velocity) = 0;
+
+  /**
+   * @brief local_planner_base isGoalReached - Check whether the robot has reached its goal, given the current pose & velocity.
+   *
+   * The pose that it checks against is the last pose in the current global plan.
+   * The calculation is delegated to the goal_checker plugin.
+   *
+   * @param pose Current pose
+   * @param velocity Current velocity
+   * @return True if the robot should be considered as having reached the goal.
+   */
+  virtual bool isGoalReached(
```

Comment:
We want to remove this function from the local planner interface. Instead, nav2_controller should load the plugin and call it directly.

Reply:
Now, I removed it from here.